invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Item/ItemForms/ItemEditor/ItemEditor.js

```diff
@@ -1,8 +1,23 @@
 import React, { Component } from 'react';
 import { Loader, Error } from '../../../../../common/components';
 import { ItemForm } from './components';
+import get from 'lodash/get';
 
 export class ItemEditor extends Component {
+  get initialData() {
+    const doc = get(this.props, 'location.state.document', null);
+    if (doc) {
+      return {
+        metadata: {
+          document: doc.metadata,
+          document_pid: doc.metadata.pid,
```

Comment:
I'm not sure why you went for changing the structure of the document metadata. Wouldn't it better to just return the doc object?

Reply:
The data that's being prepared is the item's data. So in order to prefill the item's `document` and `document_pid` fields we pass an empty item with just those two fields populated.

Reply:
May I suggest to try to improve readability somehow then if it was not super clear?
Example:
```
const openedFromDocument = 'location.state.document' in this.props;
const doc = get(this.props, 'location.state.document', null);
if (openedFromDocument && doc) {
    return prefillWithDocument(...)
}
```